src/edg.rs

```diff
@@ -417,7 +567,23 @@ impl<B: Broker<V> + Debug, G: ExtendedDependencyGraph<V> + Send + Sync + Debug,
                         },
                     );
                 }
-                VertexAssignment::UNDECIDED => {} // fallthrough
+                _ => {
+                    // update distance
+                    let dist = self.distances.get(vertex).unwrap_or(&0);
+                    self.distances
+                        .insert(vertex.clone(), max(dist.clone(), distance));
+
+                    self.mark_interest(vertex, requester);
+
+                    if let Some(dependencies) = self.depends.get(vertex) {
```

Comment:
Is it a bug that `dependencies` isn't used or could this have been replaced with
```suggestion
                    if self.depends.contains_key(vertex) {
```